src/test/java/seedu/address/logic/commands/FilterCommandTest.java

```diff
@@ -0,0 +1,94 @@
+
+import java.util.Arrays;
+import java.util.Hashtable;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.lab.Lab;
+import seedu.address.model.person.lab.LabStatus;
+import seedu.address.model.person.lab.StudentHasLabPredicate;
+
+
+
+public class FilterCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        StudentHasLabPredicate firstPredicate =
+                new StudentHasLabPredicate((new Lab("1")).thatIs(LabStatus.GRADED));
+        StudentHasLabPredicate secondPredicate =
+                new StudentHasLabPredicate((new Lab("1")).thatIs(LabStatus.UNSUBMITTED));
+        StudentHasLabPredicate thirdPredicate =
+                new StudentHasLabPredicate((new Lab("3")).thatIs(LabStatus.GRADED));
+        StudentHasLabPredicate fourthPredicate =
+                new StudentHasLabPredicate((new Lab("4")).thatIs(LabStatus.SUBMITTED));
+
+        FilterCommand firstFilterCommand = new FilterCommand(firstPredicate);
+        FilterCommand secondFilterCommand = new FilterCommand(secondPredicate);
+        FilterCommand thirdFilterCommand = new FilterCommand(thirdPredicate);
+        FilterCommand fourthFilterCommand = new FilterCommand(fourthPredicate);
+
+        // same object -> returns true
+        assertTrue(firstFilterCommand.equals(firstFilterCommand));
+
+        // same values -> returns true
+        FilterCommand firstFilterCommandCopy = new FilterCommand(firstPredicate);
+        assertTrue(firstFilterCommand.equals(firstFilterCommandCopy));
+
+        // same labNumber different labStatus -> returns false
+        assertFalse(firstFilterCommand.equals(secondFilterCommand));
+
+        // different labNumber same labStatus -> returns false
+        assertFalse(firstFilterCommand.equals(thirdFilterCommand));
+
+        // different labNumber different labStatus -> returns false
+        assertFalse(firstFilterCommand.equals(fourthFilterCommand));
+
+        // null -> returns false
+        assertFalse(firstFilterCommand.equals(null));
+    }
+
+    @Test
+    public void execute_success_multiplePersonsFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 3);
+        StudentHasLabPredicate predicate = preparePredicate("l/1 s/u");
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE, CARL, DANIEL), model.getFilteredPersonList());
+    }
+
+    private StudentHasLabPredicate preparePredicate(String userInput) {
```

Comment:
(minor thing) Perhaps want to go through the `ArgumentMultimap` class instead?

Reply:
Hmm, I was thinking of isolating the test cases, so that this test wont depend on the <code>ArgumentMultimap</code>. What do you think?

Reply:
I think that's valid too!